### PR TITLE
[FEATURE] F/great 1400/postgres constr

### PR DESCRIPTION
**Pull request by billdirks**
Update postgres datasource to validate the connection string starts with `postgresql`.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: great_expectations/experimental/datasources/postgres_datasource.py [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
+if TYPE_CHECKING:
+    PostgresConnectionString = str
+else:
+    PostgresConnectionString = constr(regex=r"^postgresql")
+
 
 class PostgresDatasource(SQLDatasource):
     type: Literal["postgres"] = "postgres"  # type: ignore[assignment]
+    connection_string: PostgresConnectionString
```
Comment (Kilo59) [inline great_expectations/experimental/datasources/postgres_datasource.py]: @billdirks did you try using the `PostgresDsn` type for this?
```python
from pydantic import PostgresDsn
```

```suggestion
    connection_string: PostgresDsn
```
https://docs.pydantic.dev/usage/types/#urls

**Review by Kilo59 — APPROVED** [review]
I think we should try to use the dedicated `PostgresDsn` type... although it's possible this could fail when we pass it to the underlying `ExecutionEngine` (if it's expecting a string).

If that's the case I'm fine with the `constr` and we could revisit this later.
I do think the errors about specific parts of the URI that are "wrong" make it valuable.

File: great_expectations/experimental/datasources/postgres_datasource.py (hunk 1 above)
Comment (billdirks) [reply great_expectations/experimental/datasources/postgres_datasource.py]: I hadn't tried this. Thanks for pointing it out! Done and I added tests.

File: tests/experimental/datasources/test_sqlite_datasource.py [hunk 2]
```diff
@@ -26,6 +27,7 @@ def test_connection_string_starts_with_sqlite():
     assert datasource.connection_string == connection_string
 
 
+@pytest.mark.unit
```
Comment (billdirks) [inline tests/experimental/datasources/test_sqlite_datasource.py:30]: I forgot to add these markers to the equivalent sqlite tests so added them now.

**Pull request merged**